sdk/formrecognizer/azure-ai-formrecognizer/azure/ai/formrecognizer/_document_analysis_client.py

```diff
@@ -123,7 +126,10 @@ def begin_analyze_document(
         if not model_id:
             raise ValueError("model_id cannot be None or empty.")
 
-        return self._client.begin_analyze_document(  # type: ignore
+        _client_op_path = self._client.document_models
+        if self._api_version == DocumentAnalysisApiVersion.V2022_08_31:
+            _client_op_path = self._client
+        return _client_op_path.begin_analyze_document(  # type: ignore
```

Comment:
API version 2022-08-31 won't be able to access the `document_models` operation group on the client so we'll need to refactor this behind an API version check. Commenting to open an issue